src/main/scala/is/hail/HailContext.scala

```diff
@@ -233,14 +233,20 @@ class HailContext private(val sc: SparkContext,
     tolerance: Double = 0.2,
     nPartitions: Option[Int] = None): VariantDataset = {
 
-    val inputs = hadoopConf.globAll(files)
+    val inputs = hadoopConf.globAll(files).flatMap { file =>
+      if (hadoopConf.isDir(file))
+        hadoopConf.glob(file + "/part-*").filter(!_.getPath.getName.endsWith(".idx")).map(_.getPath.toString)
```

Comment:
I'd filter positively for the things you're looking for, maybe something like `fs.listStatus(file).map(_.getPath.toString).filter(p => "part-[0-9]+".r.matches(p))`?

Note, there is no Regex.matches, which is super annoying.  Can you add it to utils?  I think something along the lines of:

```
class RichRegex(r: Regex) {
  def matches(s: String): Boolean = r.pattern.matcher(s).match
}
```

will work.